command/ca/certificate.go

```diff
@@ -259,10 +328,21 @@ func (f *certificateFlow) getClient(ctx *cli.Context, subject, tok string) (caCl
 	return ca.NewClient(caURL, options...)
 }
 
+type errACMEURL struct {
+	id string
+}
+
+func (e *errACMEURL) Error() string {
+	return "Unsupported method for public facing ACME flows"
+}
+
 // GenerateToken generates a token for immediate use (therefore only default
 // validity values will be used). The token is generated either with the offline
 // token flow or the online mode.
 func (f *certificateFlow) GenerateToken(ctx *cli.Context, subject string, sans []string) (string, error) {
+	if ctx.IsSet("acme") {
+		return "", &errACMEURL{}
+	}
```

Comment:
I think you should move this logic to certificateAction, if `--acme` is set, do not call `GenerateToken`